ui/de-lib/src/main/java/org/iplantc/de/client/services/impl/AnalysisServiceFacadeImpl.java

```diff
@@ -289,5 +294,40 @@ public void getAnalysisSteps(Analysis analysis, AsyncCallback<AnalysisStepsInfo>
         deServiceFacade.getServiceData(wrapper, new StringAnalaysisStepInfoConverter(callback, factory));
 
     }
+
+    @Override
+    public void shareAnalyses(AnalysisSharingRequestList request, AsyncCallback<String> callback) {
+        final String payload = AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(request)).getPayload();
+        GWT.log("analyis sharing request:" + payload);
+        String address = ANALYSES + "/" + "sharing";
+        ServiceCallWrapper wrapper = new ServiceCallWrapper(POST, address, payload);
+        deServiceFacade.getServiceData(wrapper, callback);
+    }
+
+    @Override
+    public void unshareAnalyses(AnalysisUnsharingRequestList request, AsyncCallback<String> callback) {
+        final String payload = AutoBeanCodex.encode(AutoBeanUtils.getAutoBean(request)).getPayload();
+        GWT.log("analysis un-sharing request:" + payload);
+        String address = ANALYSES + "/" + "unsharing";
+        ServiceCallWrapper wrapper = new ServiceCallWrapper(POST, address, payload);
+        deServiceFacade.getServiceData(wrapper, callback);
+    }
+
+    @Override
+    public void getPermissions(List<Analysis> analyses, AsyncCallback<String> callback) {
+        Splittable anaObj = StringQuoter.createSplittable();
+        Splittable idArr = StringQuoter.createIndexed();
+
+        for(Analysis a : analyses) {
+            Splittable item = StringQuoter.create(a.getId());
+            item.assign(idArr, idArr.size());
+        }
+
+        idArr.assign(anaObj, "analyses");
+        String address = ANALYSES + "/" + "permission-lister";
+        ServiceCallWrapper wrapper = new ServiceCallWrapper(BaseServiceCallWrapper.Type.POST, address, anaObj.getPayload());
```

Comment:
I think encoding an `AutoBean` would be better than raw `Splittable` manipulation.